catalog/core/catalog-core-solrcommands/src/main/java/org/codice/ddf/commands/solr/BackupCommand.java

```diff
@@ -116,4 +187,145 @@ private String getBackupUrl(String coreName) {
+                                i,
+                                name,
+                                value));
+                    }
+                }
+            }
+        } catch (Exception e) {
+            printErrorMessage(String.format("Backup status failed. %s", e.getMessage()));
+        }
+    }
+
+    private String getBackupName() {
+        long timestamp = System.currentTimeMillis();
+        return String.format("%s_%d", coreName, timestamp);
+    }
+
+    private void verifySingleNodeBackupInput() {
+        if (asyncBackup || asyncBackupStatus || StringUtils.isNotBlank(asyncBackupReqId)) {
+            throw new IllegalArgumentException(
+                    "Invalid Argument(s). Run solr:backup --help for usage details.");
+        }
+    }
+
+    private void verifyCloudBackupInput() {
+        if (StringUtils.isBlank(backupLocation) || asyncBackupStatus || StringUtils.isNotBlank(asyncBackupReqId) || numberToKeep > 0) {
+            throw new IllegalArgumentException(
+                    "Invalid Argument(s). Run solr:backup --help for usage details.");
+        }
+    }
+
+    private void verifyCloudBackupStatusInput() {
+        if (StringUtils.isBlank(asyncBackupReqId) || numberToKeep > 0 || StringUtils.isNotBlank(
+                backupLocation) || !StringUtils.equals(coreName, DEFAULT_CORE_NAME)
+                || asyncBackup) {
+            throw new IllegalArgumentException(
+                    "Invalid Argument(s). Run solr:backup --help for usage details.");
+        }
+    }
+
+    private void performSolrCloudBackup(SolrClient client) {
+        String backupName = getBackupName();
+        LOGGER.debug("Backing up collection {} to {} using backup name {}.",
+                coreName,
+                backupLocation,
+                backupName);
+        printInfoMessage(String.format(
+                "Backing up collection [%s] to shared location [%s] using backup name [%s].",
+                coreName,
+                backupLocation,
+                backupName));
+        if (optimizeCollection(client, coreName)) {
+            try {
+                if (asyncBackup) {
+                    String requestId = backupAsync(client, coreName, backupLocation, backupName);
+                    printInfoMessage("Solr Cloud backup request Id: " + requestId);
+                } else {
+                    boolean isSuccess = backup(client, coreName, backupLocation, backupName);
+                    if (isSuccess) {
+                        printInfoMessage("Backup complete.");
+                    } else {
+                        printErrorMessage("Backup failed.");
+                    }
+                }
+            } catch (Exception e) {
+                printErrorMessage(String.format("Backup failed. %s", e.getMessage()));
+            }
+        }
+    }
+
+    private boolean optimizeCollection(SolrClient client, String collection) {
```

Comment:
This is not a work-around.  We should optimize a collection before backup.  This method throws an exception now.